x-pack/plugins/apm/server/routes/alerts/rule_types/error_count/register_error_count_rule_type.test.ts

```diff
@@ -142,25 +241,225 @@ describe('Error count alert', () => {
       environment: 'env-foo',
       threshold: 2,
       triggerValue: 5,
-      reason: 'Error count is 5 in the last 5 mins for foo. Alert when > 2.',
+      reason:
+        'Error count is 5 in the last 5 mins for foo, env-foo, transaction name not defined. Alert when > 2.',
       interval: '5 mins',
       viewInAppUrl:
         'http://localhost:5601/eyr/app/apm/services/foo/errors?environment=env-foo',
+      'transaction.name': 'Not defined',
```

Comment:
This looks wrong. If `transaction.name` is not specified as grouping param it should just be `undefined` (not be there)

Reply:
I followed the existing pattern where if environment is not specified, the `environment` context variable had `Not defined` value. The same behaviour is now applied to all group keys. If we do not include undefined values in context variables, it will behave differently than earlier versions.

Reply:
Ah, I see what you mean. Actually, in this case, the `transaction.name` is specified in the grouping parameter, but it is not defined in the source. The bucket key will contain `TRANSACTION_NAME_NOT_DEFINED`, similar to when `environment` is not defined.

If `transaction.name` is not specified in the grouping param, then we are not adding it anywhere.